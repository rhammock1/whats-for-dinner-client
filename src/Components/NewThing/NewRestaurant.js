import React from 'react';


const NewRestaurant = function() {
  return (
    <>
      <div className='form-group'>
        <label htmlFor='title'>Restaurant Name: <span className='red'>* </span></label>
        <input type='text' id='title' name='title' required placeholder='e.g. Papi Queso' />        
      </div>
      <div className='form-group'>
        <label htmlFor='style'>Is it Chain or Local? <span className='red'>* </span></label>
        <select id='style' name='style' required>
          <option value=''>Please Choose one:</option>
          <option value='chain'>Chain</option>
          <option value='local'>Local</option>
        </select>
      </div>
      <div className='form-group'>
        <label htmlFor='phone_number'>Phone Number: </label>
        <input type='tel' id='phone_number' name='phone_number' placeholder='e.g. 123-456-7890' pattern='^(\+\d{1,2}\s)?\(?\d{3}\)?[\s.-]\d{3}[\s.-]\d{4}$'/>        
      </div>
      <div className='form-group'>
        <label htmlFor='web_url'>Website: </label>
        <input type='url' id='web_url' name='web_url' placeholder='e.g. http://cool-restaurant.com/'  />        
      </div>
      <div className='form-group'>
        <label htmlFor='restaurant_address'>Address: </label>
        <input type='text' id='restaurant_address' placeholder='e.g. 123 Main St.' name='restaurant_address' />        
      </div>
    </>
    )
}

export default NewRestaurant;