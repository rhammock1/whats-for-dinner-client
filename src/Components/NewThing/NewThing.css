input {
  background-color: #1A1A2E;
  color: #efefef;
}
.red {
  color: red;
  font-weight: bold;
}
.new-thing-form {
  width: 75%;
  height: auto;
  margin: auto;
  padding: 5px 10px;
  background-color: #313753;
  border-radius: 5px;
  box-shadow: 2px 3px 6px 3px #121212; 
}
.required {
  color:red;
  
}
input[type=text], input[type=number], select {
  padding: 5px;
}

.form-container {
  display: flex;
  width: 100%;
  flex-direction: column;
  align-items: center;
}