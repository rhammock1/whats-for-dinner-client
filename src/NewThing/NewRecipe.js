import React from 'react';


class NewRecipe extends React.Component {

 

  render() {
    return (
      <>
        <div className='form-group'>
          <label htmlFor='title'>Recipe Name: <span>* required</span></label>
          <input type='text' id='title' name='title' required placeholder='e.g. Mac N Cheese' />        
        </div>
        <div onChange={event => this.props.handleChange(event)} className='form-group'>
          <label htmlFor='amount'>Amount <span>* required</span></label>
          <input type='number' id='amount' name='amount' title='amount' placeholder='2' /> 
          <label htmlFor='unit'>Unit <span>* required</span></label>
          <input type='text' id='unit' name='unit' placeholder='Cup' /> 
          <label htmlFor='ingredient'>Ingredient <span>* required</span></label>
          <input type='text' id='ingredient' name='ingredient' placeholder='Carrots' />
          <button type='button' onClick={() => this.props.handleRemoveIngredient()}>Remove previous ingredient</button> 
          <button type='button' onClick={() => this.props.handleAddIngredient()}>Add new ingrdient</button>
        </div>
        <div className='form-group'>
          <ul className='ingredients'>
            {this.props.ingredients.map(ingredient => <li key={ingredient.id}>{ingredient.amount} {ingredient.unit} {ingredient.ingredient}</li> )}
          </ul>
        </div>
        <div className='form-group'>
          <label htmlFor='content'>Steps: <span>* required</span></label>
          <textarea id='content' name='content' placeholder='Step 1: '  />        
        </div>

      </>
    )
  }
  
}

export default NewRecipe;