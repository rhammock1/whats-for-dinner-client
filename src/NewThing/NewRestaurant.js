import React from 'react';


const NewRestaurant = function() {
  return (
    <>
      <div className='form-group'>
        <label htmlFor='title'>Restaurant Name: <span>* required</span></label>
        <input type='text' id='title' name='title' required placeholder='e.g. Papi Queso' />        
      </div>
      <div className='form-group'>
        <label htmlFor='phone-number'>Phone Number: <span>* optional</span></label>
        <input type='tel' id='phone-number' name='phone-number' placeholder='e.g. 123-456-7890' pattern='^(\+\d{1,2}\s)?\(?\d{3}\)?[\s.-]\d{3}[\s.-]\d{4}$'/>        
      </div>
      <div className='form-group'>
        <label htmlFor='website'>Website: <span>* optional</span></label>
        <input type='url' id='website' name='website' placeholder='e.g. http://cool-restaurant.com/'  />        
      </div>
      <div className='form-group'>
        <label htmlFor='address'>Address: <span>* optional</span></label>
        <input type='text' id='address' placeholder='e.g. 123 Main St.' />        
      </div>
    </>
    )
}

export default NewRestaurant;