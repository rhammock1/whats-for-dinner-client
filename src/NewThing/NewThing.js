import React from 'react';
import NewRestaurant from './NewRestaurant';

class NewThing extends React.Component {

  state = {
    thing: '',
    error: null,
  }

  handleChange = (event) => {
    this.setState({ thing: event.target.value })
  }

  handleSubmitThing = event => {
    event.preventDefault();
    console.log('hello');
    // Needs to make a post request to save the new thing with their userId
  }

  render() {
    const thingCapitalized = this.state.thing.charAt(0).toUpperCase() + this.state.thing.slice(1)
    return (
      <>
      <h2>Add New {thingCapitalized}</h2>
      <div className='form-container'>
        <form className='new-thing-form' onSubmit={event => this.handleSubmitThing(event)}>
          <fieldset>
            <legend>
              Add New:
            </legend>
            <div className='form-group'>
              <label htmlFor='new-type'>Restaurant or Recipe? </label>
              <select onChange={event => this.handleChange(event)} id='new-type' name='new-type' required>
                <option value=''>Please Choose one:</option>
                <option value='restaurant'>Restaurant</option>
                <option value='recipe'>Recipe</option>
              </select>
            </div>
          {this.state.thing === 'restaurant'
                ? <div className='form-group'><NewRestaurant /></div>
                : null}
                {/*  : this.state.thing === 'recipe' ? <div className='form-group'><NewRecipe /></div>
                 : null} */}
            <button type='submit'>Add {this.state.thing}</button>
          </fieldset>
        </form>
      </div>
      </>
        )
  }
}

export default NewThing;