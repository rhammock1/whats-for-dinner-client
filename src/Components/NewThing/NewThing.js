/* eslint-disable react/destructuring-assignment */
/* eslint-disable jsx-a11y/label-has-associated-control */
/* eslint-disable no-nested-ternary */
import React from 'react';
import apiService from '../../services/api-service';
import NewRecipe from './NewRecipe/NewRecipe';
import NewRestaurant from './NewRestaurant/NewRestaurant';
import './NewThing.css';

class NewThing extends React.Component {
  static defaultProps = {
    match: {
      params: {
        userId: 0,
      },
    },

  }

  constructor(props) {
    super(props);
    this.myRef = React.createRef();
  }

  state = {
    thing: '',
    error: null,
    amount: 0,
    unit: '',
    ingredient: '',
    ingredients: [],
    added: false,
  }

  handleAddIngredient = () => {
    const {
      amount,
      unit,
      ingredient,
      ingredients,
    } = this.state;
    const newIngredient = { amount: parseFloat(amount), unit, ingredient };
    if (newIngredient.amount === 0 || newIngredient.unit === '' || newIngredient.ingredient === '') {
      this.setState({ error: 'Please fill out ingredient fields' });
      setTimeout(() => this.setState({ error: null }), 5000);
      return;
    }
    const newIngredients = [...ingredients];
    newIngredients.push(newIngredient);

    ['amount', 'unit', 'ingredient'].forEach((input) => {
      const element = document.getElementById(input);
      element.value = '';
    });
    this.setState({
      ingredients: newIngredients, amount: 0, unit: '', ingredient: '',
    });
  }

  handleIngredientChange = (event) => {
    const { name } = event.target;

    const { value } = event.target;
    this.setState({
      [name]: value,
    });
  }

  handleChange = (event) => {
    this.setState({ thing: event.target.value });
  }

  handleRemoveIngredient = () => {
    const { ingredients } = this.state;
    ingredients.pop();
    this.setState({ ingredients });
  }

  handleSubmitThing = (event) => {
    event.preventDefault();
    const { userId } = this.props.match.params;

    let thing;
    if (this.state.thing === 'restaurant') {
      const {
        title, phone_number, web_url, restaurant_address, style,
      } = event.target;

      thing = {
        title: title.value,
        phone_number: phone_number.value,
        web_url: web_url.value,
        restaurant_address: restaurant_address.value,
        style: style.value,
        user_id: parseFloat(userId),
      };

      apiService.postNewThing('restaurants', thing)
        .then(() => this.setState({ added: true }))
        .catch((error) => this.setState({ error }));
    }
    if (this.state.thing === 'recipe') {
      const { title, content } = event.target;
      const { ingredients } = this.state;
      thing = {
        title: title.value,
        content: content.value,
        user_id: userId,
      };
      apiService.postNewThing('recipes', thing)
        .then((res) => {
          ingredients.forEach((ingredient) => ingredient.recipe_id === res.id);

          apiService.postIngredients(res.id, ingredients)
            .catch((error) => this.setState({ error }));
        })
        .then(() => {
          this.setState({ added: true });
        })
        .catch((error) => this.setState({ error }));
    }
  }

  handleSuccessClick = () => {
    this.setState({ added: false });
    const { userId } = this.props.match.params;
    this.props.history.push(`/${userId}/newThing`);
  }

  render() {
    const { error, thing, added } = this.state;
    const thingCapitalized = thing.charAt(0).toUpperCase() + thing.slice(1);
    return (
      <>
        <h2>
          Add New
          { thingCapitalized }
        </h2>
        <div role="alert">{error && <p className="red">{error}</p>}</div>
        <div className="form-container">
          <form className="new-thing-form" onSubmit={(event) => this.handleSubmitThing(event)}>
            <fieldset>
              <legend>
                Add New:
              </legend>
              <div className="form-group">
                <label htmlFor="new-type">Restaurant or Recipe? </label>
                <select onChange={(event) => this.handleChange(event)} id="new-type" name="new-type" required>
                  <option value="">Please Choose one:</option>
                  <option value="restaurant">Restaurant</option>
                  <option value="recipe">Recipe</option>
                </select>
              </div>
              {thing === 'restaurant'
                ? <div className="form-group"><NewRestaurant /></div>
                : thing === 'recipe' ? <div className="form-group"><NewRecipe refs={this.myRef} handleRemoveIngredient={this.handleRemoveIngredient} ingredients={this.state.ingredients} handleAddIngredient={this.handleAddIngredient} handleChange={this.handleIngredientChange} /></div>
                  : null}
              <button type="submit">
                Add
                {thing}
              </button>
              {thing ? <p className="required">* is required field</p> : null}
            </fieldset>
          </form>
          {added ? (
            <div className="added">
              <p>
                Successfully Added
                {thingCapitalized}
              </p>
              <button onClick={() => this.handleSuccessClick()} type="button">Back</button>
            </div>
          ) : null}
        </div>
      </>
    );
  }
}

export default NewThing;
