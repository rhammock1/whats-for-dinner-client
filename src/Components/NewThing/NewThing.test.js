import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import NewThing from './NewThing';


describe('Form component for user to add new thing', () => {
  const seedIngredients = [];
  beforeAll(() => {
    for(let i = 0; i < 5; i++) {
      seedIngredients.push(`${i} each ingredient${i}`);
    }
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter><NewThing />
    </BrowserRouter>, div)
  })
  it('renders the complete form', () => {
    const wrapper = shallow(<NewThing />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})