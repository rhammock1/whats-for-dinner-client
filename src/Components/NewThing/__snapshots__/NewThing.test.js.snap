// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component for user to add new thing renders the complete form 1`] = `
<Fragment>
  <h2>
    Add New 
  </h2>
  <div
    role="alert"
  />
  <div
    className="form-container"
  >
    <form
      className="new-thing-form"
      onSubmit={[Function]}
    >
      <fieldset>
        <legend>
          Add New:
        </legend>
        <div
          className="form-group"
        >
          <label
            htmlFor="new-type"
          >
            Restaurant or Recipe? 
          </label>
          <select
            id="new-type"
            name="new-type"
            onChange={[Function]}
            required={true}
          >
            <option
              value=""
            >
              Please Choose one:
            </option>
            <option
              value="restaurant"
            >
              Restaurant
            </option>
            <option
              value="recipe"
            >
              Recipe
            </option>
          </select>
        </div>
        <button
          type="submit"
        >
          Add 
        </button>
      </fieldset>
    </form>
  </div>
</Fragment>
`;
